### DISCUSSION TITLE
Is there any way to have "ltp -noff" recognize FFs read from a Liberty file?

### DISCUSSION THREAD

**Discussion by tcal-x** [YosysHQ/yosys#3899] [Q&A]
`ltp -noff` will break cycles at flops when operating on gate-level logic that has not been mapped to actual standard cells.   However, after mapping to a std cell library from a Liberty file, it seems that it does not recognize the library's flops, and traverses through them, and also issues voluminous warnings about finding cycles.

Is there something I'm not doing correctly in the flow?  I've tried running `read_liberty`, `read_liberty -wb`, and `read_liberty -lib` before the `ltp -noff` command, but it doesn't seem to help.

As seen with `dfflibmap`, I know that it's possible to recognize which cells in the Liberty file are FFs.

Thanks!

**Comment by whitequark** [comment]
No, it only takes into account the Yosys FF types: https://github.com/YosysHQ/yosys/blob/6405bbab1e8afc587febf41339aaf8b514c8f202/passes/cmds/ltp.cc#L45-L48
